client/src/app/components/navigation-bar/navigation-bar.component.ts

```diff
@@ -55,47 +55,47 @@ export class NavigationBarComponent implements OnInit {
 
   async discover() {
     const courseIterationId = this.allocationData.courseIteration?.id;
-    console.log('Discovering', courseIterationId);
     if (!courseIterationId) {
       return;
     }
     const serverCollaborationData = await this.websocketService.discover(courseIterationId);
-    if (!serverCollaborationData) {
+    if (!serverCollaborationData || !serverCollaborationData.allocations) {
       this.subscribe();
       return;
     }
 
-    const storedAllocations = this.allocationsService.getAllocations();
-    const storedConstraints = this.constraintsService.getConstraints();
-    const storedLockedStudents = this.lockedStudentsService.getLocks();
-    const storedCollaborationData = {
-      allocations: storedAllocations,
-      constraints: storedConstraints,
-      lockedStudents: storedLockedStudents,
-    };
-
-    if (JSON.stringify(serverCollaborationData) === JSON.stringify(storedCollaborationData)) {
+    if (
+      this.allocationsService.equals(serverCollaborationData.allocations) &&
+      this.constraintsService.equals(serverCollaborationData.constraints) &&
+      this.lockedStudentsService.equalsAsArray(serverCollaborationData.lockedStudents)
+    ) {
       this.subscribe();
       return;
     }
 
-    this.overlayService.displayComponent(ConfirmationOverlayComponent, {
-      action: 'Overwrite Allocations & Constraints',
-      actionDescription:
-        'There is a different allocation state available. Do you want to load it? This will overwrite your current allocation. Not loading it will overwrite the other allocation.',
-      onConfirmed: async () => {
-        await this.subscribe();
-        this.overlayService.closeOverlay();
-      },
-      onCancelled: async () => {
+    const overlayData = {
+      title: 'Connected to Collaboration Service',
+      description:
+        'The Collaboration Service has a different allocations and constraints state available. Do you want to load it? This will overwrite your current allocations, constraints and locked students. Not loading it will overwrite the other data. Be careful, this action cannot be undone.',
+      primaryText: 'Use Collaboration Data',
```

Comment:
Per default this button has the primary color, right? I would change the icon order and rather use the secondary color for "Use Collaboration Data"